### PR TITLE
Reference Documentation Cleanup

### PR DESCRIPTION
**Pull request by meltsufin**
This is an attempt to improve consistency and structure.
It's by no means making it perfect, but hopefully moves it
in the right direction to counter the drift.

Fixes: #2274.

### CODE REVIEW

File: docs/src/main/asciidoc/trace.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-== Spring Cloud Sleuth
+== Stackdriver Trace
```
Comment (dzou) [inline docs/src/main/asciidoc/trace.adoc]: Maybe add an additional sentence in the first paragraph that mentions stackdriver as it opens with "Spring Cloud Sleuth" while the title is Stackdriver Trace.

Comment (elefeint) [inline docs/src/main/asciidoc/trace.adoc]: Technically Cloud Trace...

Comment (meltsufin) [reply docs/src/main/asciidoc/trace.adoc]: I know, I should probably use this opportunity to rename it across the board. We just can't rename class names and module names.

Comment (meltsufin) [reply docs/src/main/asciidoc/trace.adoc]: It's actually a substantial change that needs to be done carefully.
Let's do it separately,
https://github.com/spring-cloud/spring-cloud-gcp/issues/2296

Comment (meltsufin) [reply docs/src/main/asciidoc/trace.adoc]: Done. Take a look at the commit below.

**Pull request merged**